### PR TITLE
sdk: Remove panic from RPCs 

### LINKED ISSUE #2: Panic in SDK functions
Currently we panic in many situations:
1. We forget to set an obligatory parameter. 
2. Some attribute has unexpected value (`(netmap.NodeInfo).Price()` -- we expect to get it with `FromV2` where an error is processed).
3. Something else?

While I think (1) is ok, others are not IMO. Also in (1) we could provide obligatory parameters explicitly:  hard-to-misuse API is the best. Another option is to have `New` with obligatory parameters and `Set` methods on it (private structure with public interface).

@alexvanin @KirillovDenis @carpawell @realloc @acid-ant

### PR DESCRIPTION
**Pull request by acid-ant**
Partially close #2 

Update lint config, fix lint errors

Signed-off-by: Anton Nikiforov <an.nikiforov@yadro.com>

### CODE REVIEW

**Comment by KirillovDenis** [comment]
Let's drop tests with go1.17 from github actions

**Review by KirillovDenis — DISMISSED** [review]

**Comment by alexvanin** [comment]
> Let's drop tests with go1.17 from github actions

We can also add go1.20 :smile:

**Review by alexvanin — DISMISSED** [review]

**Review by KirillovDenis — DISMISSED** [review]

**Review by alexvanin — DISMISSED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -32,15 +32,12 @@ linters:
     - revive
 
     # some default golangci-lint linters
-    - deadcode
```
Comment (fyrchik) [inline .golangci.yml:35]: Why?

File: pool/mock_test.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"crypto/ecdsa"
 	"errors"
+
 	"go.uber.org/zap"
```
Comment (fyrchik) [inline pool/mock_test.go:9]: Unrelated to this commit, could also merge this with the next line.

File: .golangci.yml (hunk 1 above)
Comment (acid-ant) [reply .golangci.yml:35]: Deprecated since v1.49.0 of golangci-lint. Replaced by `unused`.

File: pool/mock_test.go (hunk 2 above)
Comment (acid-ant) [reply pool/mock_test.go]: Reverted.

**Review by KirillovDenis — DISMISSED** [review]

File: client/accounting.go [hunk 3]
```diff
@@ -46,17 +47,17 @@ func (x ResBalanceGet) Amount() accounting.Decimal {
 // FrostFS status codes are returned as `error`, otherwise, are included
 // in the returned result structure.
 //
-// Immediately panics if parameters are set incorrectly (see PrmBalanceGet docs).
+// Returns error if parameters are set incorrectly (see PrmBalanceGet docs).
```
Comment (fyrchik) [inline client/accounting.go:50]: `an error`

File: client/client.go [hunk 4]
```diff
@@ -69,7 +69,7 @@ func (c *Client) Init(prm PrmInit) {
 // argument, otherwise context.Background() is used. Dial returns context
 // errors, see context package docs for details.
 //
-// Panics if required parameters are set incorrectly, look carefully
+// Returns error if required parameters are set incorrectly, look carefully
```
Comment (fyrchik) [inline client/client.go:72]: Same here

File: client/accounting.go [hunk 5]
```diff
@@ -46,17 +47,17 @@ func (x ResBalanceGet) Amount() accounting.Decimal {
...
 // Return statuses:
 //   - global (see Client docs).
 func (c *Client) BalanceGet(ctx context.Context, prm PrmBalanceGet) (*ResBalanceGet, error) {
 	switch {
 	case ctx == nil:
-		panic(panicMsgMissingContext)
+		return nil, errorMsgMissingContext
 	case !prm.accountSet:
-		panic("account not set")
+		return nil, errors.New("account not set")
```
Comment (fyrchik) [inline client/accounting.go:60]: What about adding a variable for this and other errors touched in this commit?

File: client/accounting.go (hunk 3 above)
Comment (acid-ant) [reply client/accounting.go]: Fixed.

File: client/client.go (hunk 4 above)
Comment (acid-ant) [reply client/client.go]: Fixed.

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        go_versions: [ '1.17.x', '1.18.x', '1.19.x' ]
+        go_versions: [ '1.18.x', '1.19.x', '1.20.x' ]
```
Comment (fyrchik) [inline .github/workflows/tests.yml:18]: This can be dropped after a rebase on master.

**Review by fyrchik — DISMISSED** [review]

**Review by fyrchik — APPROVED** [review]

**Review by dstepanov-yadro — APPROVED** [review]

**Pull request merged**